File: app/src/main/java/com/br/apigithub/MyApplication.java
package com.br.apigithub;

import android.app.Application;

import com.br.apigithub.dagger.components.DaggerRetrofitComponent;
import com.br.apigithub.dagger.components.RetrofitComponent;
import com.br.apigithub.dagger.modules.GithubModule;
import com.br.apigithub.dagger.modules.RetrofitModule;

/**
 * Created by rlima on 04/10/18.
 */

public class MyApplication extends Application {
    private static RetrofitComponent component;

    @Override
    public void onCreate() {
        super.onCreate();
        component = DaggerRetrofitComponent
                .builder()
                .retrofitModule(new RetrofitModule())
                .githubModule(new GithubModule())
                .build();
    }

    public static RetrofitComponent getComponent() {
        return component;
    }
}

File: app/src/main/java/com/concrete/desafio/MyApplication.kt
package com.concrete.desafio

import android.app.Application
import com.concrete.desafio.di.ModulosDeDependencias
import org.koin.android.ext.android.startKoin

class MyApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        startKoin(
            this,
            listOf(
                ModulosDeDependencias.moduloDeRepositorios,
                ModulosDeDependencias.moduloDePullRequest
                )
            )
    }
}